app/components/viewer/ReplayCars.tsx

```diff
@@ -0,0 +1,254 @@
+                || timeLineGlobal.followedReplay._id !== replay._id) {
+                timeLineGlobal.followedReplay = replay;
+            } else {
+                timeLineGlobal.followedReplay = null;
+            }
+            prevOnClick = Date.now();
+        }
+    };
+
+    const [opts, setOpts] = useState({
+        fontSize: 6,
+        color: '#99ccff',
+        maxWidth: 200,
+        lineHeight: 'normal',
+        letterSpacing: 0,
+        textAlign: 'left',
+        materialType: 'MeshPhongMaterial',
+        font: fonts.Comfortaa,
+        anchorX: 'center',
+        anchorY: 'middle',
+        outlineWidth: 0.5,
+        outlineColor: 'black',
+    });
+    const [rotation, setRotation] = useState([0, 0, 0, 0]);
+
+    return (
+        <>
+            <mesh
+                position={[
+                    replay.samples[sampleIndex].position.x,
+                    replay.samples[sampleIndex].position.y,
+                    replay.samples[sampleIndex].position.z]}
+                ref={mesh}
+                scale={1}
+            >
+
+                <primitive
+                    object={fbx}
+                    dispose={null}
+                    ref={stadiumCarMesh}
+                    scale={0.01}
+                    onPointerLeave={onPointerLeave}
+                    onPointerMove={onPointerMove}
+                    onClick={onClick}
+                    receiveShadow
+                    castShadow
+                />
+
+                {showInputOverlay
+                    && <InputOverlay parentRef={mesh} sampleRef={CurrentSampleRef} camera={camera} />}
+
+                <mesh
+                    ref={meshTxt}
+                >
+                    <text
+                        position-z={200}
+                        {...opts}
+                    >
+                        {opts.materialType === 'MeshPhongMaterial' ? (
+                            <meshPhongMaterial attach="material" side={DoubleSide} color={opts.color} />
+                        ) : null}
+                    </text>
+                </mesh>
+                <mesh
+                    ref={velocityRef}
+                >
+                    <sphereBufferGeometry args={[0.1, 30, 30]} attach="geometry" />
+                    <meshBasicMaterial color={replay.color} transparent opacity={0} attach="material" />
+                </mesh>
+
+            </mesh>
+        </>
+    );
+};
+
+interface ReplayCarsProps {
+    replaysData: ReplayData[];
+    timeLineGlobal: any;
+    orbitControlsRef: any;
+    showInputOverlay: boolean;
+    replayCarOpacity: number;
+    cameraMode: string;
+}
+
+export const ReplayCars = ({
+    replaysData,
+    timeLineGlobal,
+    orbitControlsRef,
+    showInputOverlay,
+    replayCarOpacity,
+    cameraMode,
+}: ReplayCarsProps): JSX.Element => {
+    const camera = useThree((state) => state.camera);
+    const fbx = useFBX('/StadiumCarWheelsSeparated.fbx');
```

Comment:
Just like in the other comment in `ReplayCar`, I think it would be better to extract the children here already, then check whether there are elements inside, and only then render the `ReplayCar`. That way you ensure that you pass some model to the `ReplayCar`. And you can also ensure that you pass something of type `THREE.Mesh` to the `ReplayCar`